internal/crd_cache.go

```diff
@@ -0,0 +1,28 @@
+package internal
+
+import (
+	"sync"
+
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+)
+
+type CustomResourceDefinitionCache struct {
+	storage *sync.Map
+}
+
+func NewCustomResourceDefinitionCache() *CustomResourceDefinitionCache {
+	return &CustomResourceDefinitionCache{storage: &sync.Map{}}
+}
+
+func (cache *CustomResourceDefinitionCache) Get(key string) *v1.CustomResourceDefinition {
```

Comment:
It's better we return back a copied value, but not a memory reference, so that CRD saved in the cache can't be changed.